examples/infill/README.md

```diff
@@ -36,6 +36,11 @@ The `infill` program offers a seamless way to interact with LLaMA models, allowi
 
 ### Example
 
+Download a CodeLlama model:
```

Comment:
Clarify that we download Codellama because it has infill support.

We can improve this by using latest CodeGemma models - I think they are smaller and also have infill support. It would be also a good exercise to verify that we support them

Reply:
I've update clarifying the usage of CodeLlama now :+1:  

Regarding CodeGemma, I tried out a smaller model [`codegemma-2b-f16.gguf`](https://huggingface.co/google/codegemma-2b-GGUF/blob/main/codegemma-2b-f16.gguf) which did not produce a good result. I'll try a larger model, [codegemma-7b-it-f16.gguf](https://huggingface.co/google/codegemma-7b-it-GGUF/blob/main/codegemma-7b-it-f16.gguf) and see if it works (if I can run it). One thing I've noticed is that these are [models](https://huggingface.co/google/codegemma-7b-it-GGUF) are gated so I was not able to use the `hf` script to download them but had to manually download them.

Reply:
Ah yes, this should be fixed - likely we need to add these special tokens in the GGUF meta data and start using them. Could you verify that using the correct token ids (by hardcoding them) produces correct results?

Reply:
> Could you verify that using the correct token ids (by hardcoding them) produces correct results?

The following is the output when using the special token ids from CodeGemma:
```console
./infill -t 10 -ngl 15 -m models/codegemma-7b-it-f16.gguf -c 4096 --temp 0 --repeat_penalty 1.0 -n 20 --in-prefix "def helloworld():\n    print(\"hell" --in-suffix "\n   print(\"goodbye world\")\n    "
...

#####  Infill mode  #####

<|fim_prefix|> def helloworld():\n    print("hell<|fim_suffix|> \n   print("goodbye world")\n    <|fim_middle|>
o world")\n   print("goodbye world")\n   print("hello world")\<|file_separator|>

```
This looks better I think and more inline with that the CodeLlama model outputs. 
I'd be happy to take a stab at adding these special tokens to the GGUF meta data, if that is alright?

Reply:
Need to add start adding these in the GGUF meta and then using it inside `llama.cpp`.

Btw, the example that you posted seems kind of ok, but I would have expected it to after finishing the hello world print to create a new function `def goodbyeworld():`. There might be some other issues still

Reply:
> Need to add these in the GGUF meta and then use inside `llama.cpp`.

Could you give me a few pointer as to where this meta data is added? Is it added to gguf.py, like in vocab.py simliar to the other special tokens like `bos`, `eos`, `unk` etc?

> There might be some other issues still

I'll take a closer look at this as part of adding the metadata and see if I can figure out what is wrong.

Reply:
I've opened https://github.com/ggerganov/llama.cpp/pull/6689 with a suggestion.